src/shared/Microsoft.Git.CredentialManager/GitConfiguration.cs

```diff
@@ -101,6 +101,8 @@ public void Enumerate(GitConfigurationEnumerationCallback cb)
             using (Process git = _git.CreateProcess($"config --null {level} --list"))
             {
                 git.Start();
+                // To avoid deadlocks, always read the output stream first and then wait.
```

Comment:
Could you also keep the `TODO` from before? We still want to track work to stream the data instead of creating one giant `string`.

Reply:
Sure!

Reply:
Done!